acts

```diff
@@ -180,40 +187,35 @@ if [ "$backuprc" != "0" ]; then
 fi
 
-for dir in $backuptargets; do
-    log_debug "message=\"Checking $dir for old backups to delete\""
-    nicedirname=$(echo "$dir" | tr -d '/')
-    # We don't delete any yearly backups
-
-    # We keep 12 monthly backups
-    monthlybackups=$(echo "$archives" | grep "$hostname-monthly-.+-$nicedirname$" | sort -rn)
-    if [ "$(echo "$monthlybackups" | wc -l)" -gt 12 ]; then
-        log_debug 'message="More than 12 monthly backups, deleting the oldest"'
-        echo "$monthlybackups" | tail -n +13 | while read -r archiveprefixtodel; do
-            log_verbose "message=\"Deleting backup prefix $archiveprefixtodel*\""
-            echo "$archives" | grep -E "^$archiveprefixtodel" | while read -r archivetodel; do
-                log_debug "message=\"Deleting backup $archivetodel\""
-                $tarsnap -d -f "$archivetodel"
+prune_backups() {
+    backuplevel=$1 max=$2
+
+    if [ "$max" -gt 0 ]; then
+        backupslist=$(echo "$archives" | grep "$hostname-$backuplevel-.+$nicedirname$" | sort -rn)
+        if [ "$(echo "$backupslist" | wc -l)" -gt "$max" ]; then
+            log_debug "message=\"More than $max yearly backups, deleting the oldest\""
+            echo "$backupslist" | tail -n +"$max" | while read -r archiveprefixtodel; do
+                log_verbose "message=\"Deleting backup prefix $archiveprefixtodel*\""
+                echo "$archives" | grep -E "^$archiveprefixtodel" | while read -r archivetodel; do
+                    log_debug "message=\"Deleting backup $archivetodel\""
+                    $tarsnap -d -f "$archivetodel"
+                done
             done
-        done
+        else
+            log_debug "message=\"Found $(echo "$backupslist" | wc -l) $backuplevel backups, not deleting\""
+        fi
     else
-        log_debug "message=\"Found $(echo "$monthlybackups" | wc -l) monthly backups, not deleting\""
+        log_debug "message=\"Keeping all $backuplevel backups indefinitely; not deleting any\""
     fi
+}
 
-    # We keep 31 daily backups
-    dailybackups=$(echo "$archives" | grep "$hostname-daily-.+-$nicedirname$" | sort -rn)
-    if [ "$(echo "$dailybackups" | wc -l)" -gt 31 ]; then
-        log_debug "message=\"More than 30 daily backups, deleting the oldest\""
-        echo "$dailybackups" | tail -n +32 | while read -r archiveprefixtodel; do
-            log_verbose "message=\"Deleting backup prefix $archiveprefixtodel*\""
-            echo "$archives" | grep -E "^$archiveprefixtodel" | while read -r archivetodel; do
-                log_debug "message=\"Deleting backup $archivetodel\""
-                $tarsnap -d -f "$archivetodel"
-            done
-        done
-    else
-        log_debug "message=\"Found $(echo "$dailybackups" | wc -l) daily backups, not deleting any\""
-    fi
+for dir in $backuptargets; do
+    log_debug "message=\"Checking $dir for old backups to delete\""
+    nicedirname=${dir:+-${dir#/}}  # /foo -> -foo, / -> ''
```

Comment:
is this local variable no longer used?

Reply:
Are you referring to `$nicedirname`? The `prune_backups` function picks it up.

Reply:
yeah. I misread and thought it was being set twice, I think